File: nats/src/test/groovy/com/github/grimmjo/micronaut/nats/docs/quickstart/ProductClient.java
package com.github.grimmjo.micronaut.nats.docs.quickstart;

// tag::imports[]
import com.github.grimmjo.micronaut.nats.annotation.NatsClient;
import com.github.grimmjo.micronaut.nats.annotation.Subject;
import io.micronaut.messaging.annotation.Body;
// end::imports[]

// tag::clazz[]
@NatsClient // <1>
public interface ProductClient {

    @Subject("abc") // <2>
    void send(@Body byte[] data); // <3>

}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/consumer/types/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

from java.lang import Long
from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.messaging.annotation import MessageHeader
from micronaut.nats.annotation import NatsClient, Subject
# end::imports[]


@Requires(property="spec.name", value="TypeBindingSpec")
# tag::clazz[]
@NatsClient  # <1>
class ProductClient(ABC):

    @Subject("product")  # <2>
    @abstractmethod
    def send(self, data: bytes, count: Annotated[Long, MessageHeader("x-product-count")]) -> None:  # <3>
        ...
# end::clazz[]
